core/lib/dal/src/connection/mod.rs

```diff
@@ -51,41 +40,28 @@ impl ConnectionPoolBuilder {
 
     /// Builds a connection pool from this builder.
     pub async fn build(&self) -> anyhow::Result<ConnectionPool> {
-        let database_url = match self.db {
-            DbVariant::Master => get_master_database_url()?,
-            DbVariant::Replica => get_replica_database_url()?,
-            DbVariant::Prover => get_prover_database_url()?,
-            DbVariant::TestTmp => create_test_db()
-                .await
-                .context("create_test_db()")?
-                .to_string(),
-        };
-        self.build_inner(&database_url)
-            .await
-            .context("build_inner()")
-    }
-
-    pub async fn build_inner(&self, database_url: &str) -> anyhow::Result<ConnectionPool> {
-        let max_connections = self
-            .max_size
-            .unwrap_or_else(|| parse_env("DATABASE_POOL_SIZE"));
-
-        let options = PgPoolOptions::new().max_connections(max_connections);
-        let mut connect_options: PgConnectOptions = database_url
+        let options = PgPoolOptions::new().max_connections(self.max_size);
+        let mut connect_options: PgConnectOptions = self
+            .database_url
             .parse()
-            .with_context(|| format!("Failed parsing {:?} database URL", self.db))?;
+            .with_context(|| format!("Failed parsing database URL: {}", self.database_url))?;
```

Comment:
`database_url` can contain sensitive information (user / password for the database); that's why it wasn't included into the error messages previously. For the same reason, it may make sense to exclude it from `Debug` implementation of `ConnectionPoolBuilder`.